code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -114,4 +114,85 @@
 	recoil = 2 // ow my fucking hand
 	accuracy = -1 // smooth bore + short barrel = shit accuracy
 	ammo_type = /obj/item/ammo_casing/shotgun
-	// ToDo: Remove accuracy debuf in exchange for slightly injuring your hand every time you fire it.
\ No newline at end of file
+	// ToDo: Remove accuracy debuf in exchange for slightly injuring your hand every time you fire it.
+
+/obj/item/weapon/gun/projectile/revolver/lemat
+	name = "LeMat Revolver"
+	desc = "The LeMat Revolver is a 9 shot revolver with a secondary firing barrel loading shotgun shells. For when you really need something dead."
+	icon_state = "lemat"
+	item_state = "revolver"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
+	handle_casings = CYCLE_CASINGS
+	max_shells = 9
+	caliber = "38"
+	ammo_type = /obj/item/ammo_casing/c38
+	var/secondary_max_shells = 1
+	var/secondary_caliber = "shotgun"
+	var/secondary_ammo_type = /obj/item/ammo_casing/shotgun
+	var/flipped_firing = 0
+	var/list/secondary_loaded = list()
+	var/list/tertiary_loaded = list()

```

Comment:
Kinda snowflake but it's a third list used to hold the loaded ammo when swapping between the lists.   It's used in swap_firingmode() to entirely swap between the two different kinds of ammo loaded.